src/libraries/Common/tests/System/Security/Cryptography/CngKeyWrapper.cs

```diff
@@ -7,20 +7,24 @@
 
 namespace Test.Cryptography
 {
-    internal sealed class CngPlatformProviderKey : IDisposable
+    internal sealed class CngKeyWrapper : IDisposable
     {
-        public CngPlatformProviderKey(
+        public CngKeyWrapper(
             CngAlgorithm algorithm,
             string keySuffix = null,
             [CallerMemberName] string testName = null,
             params CngProperty[] additionalParameters)
+            : this(algorithm, CreateDefaultParameters(), keySuffix, testName, additionalParameters)
         {
-            CngKeyCreationParameters cngCreationParameters = new CngKeyCreationParameters
-            {
-                Provider = CngProvider.MicrosoftPlatformCryptoProvider,
-                KeyCreationOptions = CngKeyCreationOptions.OverwriteExistingKey,
-            };
+        }
 
+        public CngKeyWrapper(
+            CngAlgorithm algorithm,
+            CngKeyCreationParameters cngCreationParameters,
+            string keySuffix = null,
+            [CallerMemberName] string testName = null,
+            params CngProperty[] additionalParameters)
```

Comment:
The constructor still modifies `cngCreationParameters` with `additionalParameters`.  This is unintuitive, and sets us up for hard to diagnose failures in tests.

To quote [myself from before the file got renamed](https://github.com/dotnet/runtime/pull/102495#discussion_r1643475378):

> This ctor should be removed, and replaced with a static method for using the software KSP. E.g. `public static SelfDeletingCngKey CreateSoftwareKey`, which takes whatever options you need. That'll probably require making a private ctor that takes the prebuilt CngKeyCreationParameters; leave the existing one alone.